### PR TITLE
[search] addr:place matching.

### PR DESCRIPTION
**Pull request by vng**
Should review and test addresses described in the issues below. Should work in 90%, remains some very exotic cases.

https://github.com/organicmaps/organicmaps/issues/2689: Mannheim works; HOPFENBURG is not, because of place without actual tag place=*
Closes https://github.com/organicmaps/organicmaps/issues/5075
Closes https://github.com/organicmaps/organicmaps/issues/5890
https://github.com/organicmaps/organicmaps/issues/557: Everything should work, but there are problems with duplicating "СНТ Вишенка"
Closes https://github.com/organicmaps/organicmaps/issues/579
Closes https://github.com/organicmaps/organicmaps/issues/603

Meta issue: https://github.com/organicmaps/organicmaps/issues/5307

Update wiki here: https://github.com/organicmaps/organicmaps/issues/557#issuecomment-1732262276

### CODE REVIEW

**Comment by biodranik** [comment]
@Jean-BaptisteC can you please help with issues testing and mark which ones can be closed?

File: search/features_layer_matcher.hpp [hunk 1]
```diff
@@ -387,15 +393,60 @@ class FeaturesLayerMatcher
...
+    auto const houseNumberFilter = [&](uint32_t houseId)
+    {
+      ++numFilterInvocations;
+      if ((numFilterInvocations & 0xFF) == 0)
+        BailIfCancelled();
+
+      if (m_postcodes && !m_postcodes->HasBit(houseId))
+        return false;
+
+      /// @todo Add house -> number cache for this and MatchBuildingsWithStreets?
```
Comment (biodranik) [inline search/features_layer_matcher.hpp:427]: Did the search slow down after these commits?

**Review by biodranik — COMMENTED** [review]
1.
```
 TID(1) ASSERT FAILED
search/geocoder.cpp:1663
CHECK(resID != kInvalidId) 
Assertion failed: (false), function MatchWorld2Country, file geocoder.cpp, line 1663.
```

2. How are `addr:place` addresses displayed in the PP/UI?

3. What about the editor? Let's create/note the remaining work in some issues.
4.

**Comment by vng** [comment]
2. Displayed the same like with streets
3. We have an issue, will disable streets editing

**Comment by biodranik** [comment]
Do you mean disable editing streets completely? Let's describe the issue in detail separately. We need to find a good solution to:
1. Avoid breaking existing, properly mapped addresses.
2. Allow adding missing addresses in the right way.

**Comment by vng** [comment]
Disable address street editing _if it has addr:place_ reference. Show this reference grayed.

**Comment by biodranik** [comment]
> Disable address street editing if it has addr:place reference. Show this reference grayed.

What about new addresses? Buildings that are yet without addr:place and without addr:street, near buildings with addr:place?

**Comment by vng** [comment]
Do you have an idea how to explain the user addr:street vs addr:place?

**Comment by biodranik** [comment]
@vng please create a separate issue and describe the context in details. I'll help with the wording.

**Comment by vng** [comment]
Already have one. Can discuss there https://github.com/organicmaps/organicmaps/issues/6491

**Review by biodranik — APPROVED** [review]
LGTM if PP and search results display the address properly.

Editor-related issues can be fixed separately.

**Pull request merged**